app/assets/javascripts/application.js.erb

```diff
@@ -230,6 +130,60 @@ class _LobstersFunction {
     document.location = "/login?return=" + encodeURIComponent(document.location);
   }
 
+  modalFlaggingDropDown(flaggedItemType, voterEl) {
+    if (!Lobsters.curUser)
+      return Lobster.bounceToLogin();
+
+      const li = parentSelector(voterEl, '.story, .comment');
+      if (li.classList.contains('flagged')) {
+        /* already upvoted, neutralize */
+        if (li.classList.contains('story')) {
+        Lobster.voteStory(voterEl, -1, null);
+        } else {
+          Lobster.voteComment(voterEl, -1, null);
+        }
+        return
+      }
+
+    if (document.querySelector("#flag_dropdown") || document.querySelector('#modal_backdrop')) {
+      Lobster.modalRemoval()
+    }
+
+    const modalDiv = document.createElement("div");
+    modalDiv.setAttribute('id', 'modal_backdrop');
+    document.body.appendChild(modalDiv);
+
+    const flaggingDropDown = document.createElement('div');
+    flaggingDropDown.setAttribute('id', 'flag_dropdown');
+    const flaggingWrapper = document.createElement('span');
+    flaggingWrapper.setAttribute('class', 'flag-wrapper')
+    flaggingWrapper.style.position = 'relative';
+    flaggingWrapper.style.maxWidth = '1 px';
+    flaggingWrapper.style.maxHeight = '1 px';
+    flaggingWrapper.appendChild(flaggingDropDown);
+    voterEl.after(flaggingWrapper);
+
+    let reasons;
+    if (flaggedItemType == "comment") {
+      reasons = Lobster.commentFlagReasons;
+    } else {
+      reasons = Lobster.storyFlagReasons;
+    }
+
+    Object.keys(reasons).map(function(k, v) {
+      let a = document.createElement('a')
+      a.textContent = reasons[k]
+      a.setAttribute('data', k)
+      if (k === ''){
+        a.setAttribute('href', '#' )
```

Comment:
This line is the same in both branches of the `if/else`, so move it above the `if`